server/controllers/usersController.api.js

```diff
@@ -1,21 +1,49 @@
 'use strict';
 const { User } = require('../models');
+const upload = require('../services/file-upload');
+const singleUpload = upload.single('image');
 
 exports.getUser = (req, res) => {
   const { id } = req.body;
   User.findOne({ id }, (err, user) => {
     if (err) {
       return res.status(400).json({ message: 'an error occurred' });
     } else if (user) {
-      const { id, name, email } = user;
-      return res.json({
-        status: 200,
+      const { id, name, email, location, about, profilePic } = user;
+      return res.status(200).json({
         id,
         name,
-        email
+        email,
+        location,
+        about,
+        profilePic
       });
     } else {
       return res.status(400).json({ message: 'an error occurred' });
     }
   });
 };
+
+exports.editUser = (req, res) => {
+  singleUpload(req, res, function (err) {
+    if (err) return res.status(422).json({ errors: [{ title: 'File Upload Error', detail: err.message }] });
+
+    const { name, location, about, profilePic: newProfilePic } = req.body;
+    const profilePic = req.file ? req.file.location : newProfilePic;
+    const newInfo = { name, location, about, profilePic };
+    User.findOneAndUpdate(
```

Comment:
Looks like you are searching in the db for the same user 2x. first at userById and then here?

Reply:
Also, might be unnecessarily populating projects.

Reply:
I just removed userById in user routes. It's still being used in projects though which is fine. I also modified the isAuth to check if the token id matches the param id for security reasons. Should be just getting one request now. I also changed the get user route to use your getUsers method. Before it was just using the profile from userById

Reply:
Cool. I'm going to steal your strategy for this going forward.